File: mobile/ios/OTravelz/Networking/DTO/TransitDTOs.swift
import Foundation

public struct TransitRouteDTO: Codable, Sendable {
    public let id: String
    public let routeNumber: String?
    public let name: String?
    public let operatorName: String?
    public let origin: String?
    public let destination: String?
    public let via: [String]?
    public let stopsCount: Int?

    enum CodingKeys: String, CodingKey {
        case id
        case routeNumber = "route_number"
        case name
        case operatorName = "operator"
        case origin
        case destination
        case via
        case stopsCount = "stops_count"
    }
}

public struct RouteListDTO: Codable, Sendable {
    public let total: Int
    public let limit: Int
    public let offset: Int
    public let routes: [TransitRouteDTO]
}

public struct RouteGeometryDTO: Codable, Sendable {
    public let routeId: String
    public let coordinates: [[Double]]

    enum CodingKeys: String, CodingKey {
        case routeId = "route_id"
        case coordinates
    }
}

public struct StopNearbyDTO: Codable, Sendable {
    public let stopId: String
    public let name: String
    public let publishedName: String?
    public let canonicalStopId: String?
    public let city: String?
    public let district: String?
    public let locality: String?
    public let latitude: Double?
    public let longitude: Double?
    public let coordinateStatus: String?
    public let distanceM: Double?
    public let walkingEstimateMins: Int?
    public let routesServingStop: [String]?

    enum CodingKeys: String, CodingKey {
        case stopId = "stop_id"
        case name
        case publishedName = "published_name"
        case canonicalStopId = "canonical_stop_id"
        case city
        case district
        case locality
        case latitude
        case longitude
        case coordinateStatus = "coordinate_status"
        case distanceM = "distance_m"
        case walkingEstimateMins = "walking_estimate_mins"
        case routesServingStop = "routes_serving_stop"
    }
}

File: mobile/android/src/main/kotlin/com/otravelz/android/data/network/dto/TransitDtos.kt
package com.otravelz.android.data.network.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class TransitRouteDto(
    val id: String,
    @SerialName("route_number") val routeNumber: String? = null,
    val name: String? = null,
    val operator: String? = null,
    val origin: String? = null,
    val destination: String? = null,
    val via: List<String>? = null,
    @SerialName("stops_count") val stopsCount: Int? = null,
    @SerialName("total_distance_km") val totalDistanceKm: Double? = null
)

@Serializable
data class RouteListDto(
    val total: Int = 0,
    val limit: Int = 0,
    val offset: Int = 0,
    val routes: List<TransitRouteDto> = emptyList()
)

@Serializable
data class RouteGeometryDto(
    @SerialName("route_id") val routeId: String,
    val coordinates: List<List<Double>> = emptyList()
)

@Serializable
data class StopNearbyDto(
    @SerialName("stop_id") val stopId: String,
    val name: String,
    @SerialName("published_name") val publishedName: String? = null,
    @SerialName("canonical_stop_id") val canonicalStopId: String? = null,
    val city: String? = null,
    val district: String? = null,
    val locality: String? = null,
    val latitude: Double? = null,
    val longitude: Double? = null,
    @SerialName("coordinate_status") val coordinateStatus: String? = null,
    @SerialName("distance_m") val distanceM: Double? = null,
    @SerialName("walking_estimate_mins") val walkingEstimateMins: Int? = null,
    @SerialName("routes_serving_stop") val routesServingStop: List<String> = emptyList()
)
